observations/obs_converters/Ameriflux/fluxnetfull_to_obs.f90

```diff
@@ -0,0 +1,1180 @@
+tower%recoDTUNC84 =           values(tower%recoDTUNC84index-time_adjust)
+tower%recoNTUNC16 =           values(tower%recoNTUNC16index-time_adjust)
+tower%recoNTUNC84 =           values(tower%recoNTUNC84index-time_adjust)
+deallocate(values)
+
+
+
+read(tower%start_time(1:12), fmt='(i4, 4i2)') yeara,montha,daya,houra,mina 
+read(tower%end_time(1:12),   fmt='(i4, 4i2)') yearb,monthb,dayb,hourb,minb
+
+
+! Certain time resolutions (MM) does not define days
+if (time_resolution == 'MM') then
+   daya = 1
+   dayb = 1
+endif
+
+
+
+write(*,*)''
+write(string1, *) 'Display tower%start_time,yeara,montha,daya,houra,mina (LTC)  =', tower%start_time, yeara, montha, daya, houra, mina
+write(string2, *) 'Display tower%end_time,yearb,monthb,dayb,hourb,minb   (LTC)  =', tower%end_time, yearb, monthb, dayb, hourb, minb
+if (verbose) call error_handler(E_MSG,'Fluxnetfull_to_obs',string1,text2=string2)
+
+
+
+date_start= set_date(yeara,montha,daya,houra,mina,0)
+date_end=   set_date(yearb,monthb,dayb,hourb,minb,0)
+
+
+
+! Assign average of flux time window to obs_seq (DART time)
+tower%time_obs = (date_start+date_end) / 2
+
+! Covert from Fluxnet provided LTC to UTC, UTC is standard for DART and CLM 
+! For example, EST = UTC-5 
+if (timezoneoffset < 0.0_r8) then
+   tower%time_obs = tower%time_obs + offset
+else
+   tower%time_obs = tower%time_obs - offset
+endif
+
+
+! Reject NEE data where neeQC is missing
+if (tower%neeQC < 0) tower%neeQC = maxgoodqc + 100
+
+! If NEE (DD,WW,MM) convert from  % filled QC to integer QC  good/fair/poor
+if (time_resolution == 'DD' .or. time_resolution == 'MM' .or. &
+    time_resolution == 'WW') then
+   if (tower%neeQCfrac >= 0.90) tower%neeQC = 1 ! >90% fill is good
+   if (tower%neeQCfrac < 0.90 .and. tower%neeQC >= 0.60) tower%neeQC = 2 ! 60-90% fill is fair
+   if (tower%neeQCfrac < 0.60 .and. tower%neeQC >= 0.0_r8) tower%neeQC = 3 ! <60% fill is poor
+endif
+
+
+! The QC values are typically missing for le and h (-9999)
+! Thus  manually assign poor QC values in these cases
+if (energy_balance .eqv. .false.) then
+   if (tower%leQC  < 0.0_r8) tower%leQC  = 2
+   if (tower%hQC   < 0.0_r8) tower%hQC   = 2
+else  ! No QC values for energy balance corrected le and h.  Assign fair QC.
+   tower%leQC  = 2
+   tower%leQC  = 2
+endif
+
+
+! No qc values for gpp/reco, thus assign fair qc unless
+! the gpp/reco values are negative values, then reject.
+tower%gppNTQC = 2
+tower%gppDTQC = 2
+tower%recoNTQC = 2
+tower%recoDTQC = 2
+
+if (tower%gppNT < 0.0_r8) tower%gppNTQC = maxgoodqc + 100
+if (tower%gppDT < 0.0_r8) tower%gppDTQC = maxgoodqc + 100
+if (tower%recoNT < 0.0_r8) tower%recoNTQC = maxgoodqc + 100
+if (tower%recoDT < 0.0_r8) tower%recoDTQC = maxgoodqc + 100
+
+! Assign very bad qc to gap_filled data if user requests it
+! such that maxgoodqc threshold does not add gap_filled data to obs_seq file
+! If leQC and hQC are missing no way to identify if gap_filled, thus rejected
+if (gap_filled .eqv. .false.) then
```

Comment:
gap_filled=false means the nee, le and h are measured instead of "filled" by some methods. I assume GPP and reco in this case are partitioned from the measured nee. Then why the QC of gpp and reco are assigned bad (maxgoodqc + 100)?